homeassistant/components/water_heater/group.py

```diff
@@ -0,0 +1,34 @@
+"""Describe group states."""
+
+
+from homeassistant.components.group import GroupIntegrationRegistry
+from homeassistant.const import STATE_OFF
+from homeassistant.core import callback
+from homeassistant.helpers.typing import HomeAssistantType
+
+from . import (
+    STATE_ECO,
+    STATE_ELECTRIC,
+    STATE_GAS,
+    STATE_HEAT_PUMP,
+    STATE_HIGH_DEMAND,
+    STATE_PERFORMANCE,
+)
+
+
+@callback
+def async_describe_on_off_states(
+    hass: HomeAssistantType, registry: GroupIntegrationRegistry
+) -> None:
+    """Describe group on off states."""
+    registry.on_off_states(
+        [
+            STATE_ECO,
+            STATE_ELECTRIC,
+            STATE_PERFORMANCE,
+            STATE_HIGH_DEMAND,
+            STATE_HEAT_PUMP,
+            STATE_GAS,
+        ],
```

Comment:
Not sure if it matters for speed, but should we make the on states parameter a set everywhere? 
```suggestion
        {
            STATE_ECO,
            STATE_ELECTRIC,
            STATE_PERFORMANCE,
            STATE_HIGH_DEMAND,
            STATE_HEAT_PUMP,
            STATE_GAS,
        },
```